sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoderGenerator.java

```diff
@@ -101,12 +101,12 @@
   private static final Map<TypeName, StackManipulation> CODER_MAP;
 
   // Cache for Coder class that are already generated.
-  private static Map<UUID, Coder<Row>> generatedCoders = Maps.newHashMap();
+  private static Map<UUID, Coder<Row>> generatedCoders = Maps.newConcurrentMap();
```

Comment:
Does it cause errors? It looks like it is only modified in static initialization block, which should happen once per class loader and be thread safe.

Reply:
In my experience, even in these cases it's best to use a lock or a thread-safe data structure. If not, weird things can still go wrong (e.g. since no memory fence is set, the compiler or the processor is free to reorder the read and the write in such a way that breaks your assumptions).